scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+
+            - ``'formula'``: use a formula for the median
+            - ``'icdf'``: evaluate the inverse CDF of 0.5
+
+            Not all `method` options are available for all distributions.
+            If the selected `method` is not available, a ``NotImplementedError``
+            will be raised.
+
+        Returns
+        -------
+        out : Array
+            The median
+
+        See Also
+        --------
+        ContinuousDistribution.mean
+        ContinuousDistribution.mode
+        ContinuousDistribution.icdf
+
+        Examples
+        --------
+        Instantiate a distribution with the desired parameters:
+
+        >>> from scipy import stats
+        >>> X = stats.Uniform(a=0., b=10.)
+
+        Evaluate the inverse CDF at the desired argument:
+
+        >>> X.median()
+        5
+        >>> X.median() == X.icdf(0.5) == X.iccdf(0.5)
+        True
+
+        """
+        return self._median_dispatch(method=method, **self._parameters)
+
+    @_dispatch
+    def _median_dispatch(self, method=None, **kwargs):
+        if self._overrides('_median_formula'):
+            method = self._median_formula
+        else:
+            method = self._median_icdf
+        return method
+
+    def _median_formula(self, **kwargs):
+        raise NotImplementedError(self._not_implemented)
+
+    def _median_icdf(self, **kwargs):
+        return self._icdf_dispatch(0.5, **kwargs)
+
+    @_set_invalid_nan_property
+    def mode(self, *, method=None):
+        r"""Mode
+
+        Informally, the mode is a value that a random variable has the highest
+        probability (density) of assuming. That is, the mode is the element of
+        the support :math:`\chi` that maximizes the probability density
+        function :math:`f(x)`:
+
+        .. math::
+
+            \text{mode} = \argmax_{\chi} f(x)
+
+        Parameters
+        ----------
+        method : {None, 'formula', 'optimization'}
+            The strategy used to evaluate the mode.
+            By default (``None``), the infrastructure chooses between the
+            following options.
+
+            - ``'formula'``: use a formula for the median
+            - ``'optimization'``: numerically maximize the PDF
+
+            Not all `method` options are available for all distributions.
+            If the selected `method` is not available, a ``NotImplementedError``
+            will be raised.
+
+        Returns
+        -------
+        out : Array
+            The mode
+
+        See Also
+        --------
+        ContinuousDistribution.mean
+        ContinuousDistribution.median
+        ContinuousDistribution.pdf
+
+        Notes
+        -----
+        For some distributions
+
+        #. the mode is not unique (e.g. the uniform distribution);
+        #. the PDF has one or more singularities, and it is debateable whether
+           a singularity is considered to be in the domain and called the mode
+           (e.g. the beta distribution); and/or
```

Comment:
easier: pdf of a Gamma distribution with a < 1 goes to inf as x to 0
(beta has two parameters and has singularities if either a or b < 1)

Reply:
so you want to replace "the beta distribution" with "the gamma distribution" with certain parameters?

Reply:
it was just a suggestion to make it clearer since the beta distribution has two parameters a and b and there is no / one / two singularities depending on how you choose a and b.

Reply:
Done in next commit.

Reply:
can be resolved